openedx/core/djangoapps/util/user_messages.py

```diff
@@ -181,12 +184,47 @@ def _create_user_message(message):
         return (_create_user_message(message) for message in django_messages if self.get_namespace() in message.tags)
 
 
+# .. toggle_name: GLOBAL_NOTICE_ENABLED
+# .. toggle_implementation: SettingToggle
+# .. toggle_default: False
+# .. toggle_description: When enabled, show the contents of GLOBAL_NOTICE_MESSAGE as a message on every page. This is intended to be used as a way of communicating an upcoming or currently active maintenance window or to warn of known site issues. HTML is not supported for the message content. Message styling can be controlled with GLOBAL_NOTICE_TYPE, set to one of INFO, SUCCESS, WARNING, or ERROR (defaulting to INFO). Also see openedx.core.djangoapps.util.maintenance_banner.add_maintenance_banner for a variation that only shows a message on specific views.
+# .. toggle_use_cases: open_edx
+# .. toggle_creation_date: 2021-09-07
+GLOBAL_NOTICE_ENABLED = SettingToggle('GLOBAL_NOTICE_ENABLED', default=False)
+
+
 class PageLevelMessages(UserMessageCollection):
     """
     This set of messages appears as top page level messages.
     """
     NAMESPACE = 'page_level_messages'
 
+    @classmethod
+    def user_messages(cls, request):
+        """
+        Returns outstanding user messages, along with any persistent site-wide messages.
+        """
+        # If there is a global notice, add it to the messages just
+        # prior to retrieval. (Kind of a hack, but this ensures both
+        # code reuse and only-once-per-page display.)
+        try:
+            if GLOBAL_NOTICE_ENABLED.is_enabled():
+                # FIXME: user_messages calls the messages "body_html"
+                # internally but the messages appear to be
+                # HTML-encoded or tag-stripped in practice. Until that
+                # ambiguity is resolved, best bet is to use plaintext
+                # that would be safe either as text or HTML.
```

Comment:
I think I can wrap the content in an `HTML` call and have it be unescaped, based on what I see in the tests. Worth changing this to `GLOBAL_NOTICE_HTML` and allowing that functionality? Or should this be wrapped in `Text` and specified as plaintext only?

Reply:
I think leaving it as plaintext-only makes more sense. The previous functionality was probably written before we had more comprehensive XSS handling.

Reply:
OK, I'll wrap it in Text. I was thinking it might be nice to be able to render a link, but it's probably better to leave it less powerful.